### DISCUSSION TITLE
Better, more descriptive error messages, to simplify the learning curve for .NET beginners and students

### DISCUSSION THREAD

**Discussion by rarera** [dotnet/roslyn#69961] [Ideas]
Recently I looked into Rust, and I realized that the error messages there are very clear.

Even though I myself do not run into many error messages after a decade of C# programming, and even if quickly understand the messages, we have to put us in the shoes of beginners.

Maybe we should put a community page together, where we have all possible error messages, a try by Chat GPT to formulate them in a more readable manner, and then accept community proposals, and finally maybe allow voting in a poll to really get to better error messages with the help of the community, as I do not think this is something a few devs can do themselves.

Let's make more approachable for beginners and increase its popularity, and also stay competitive against new and upcoming languages.

**Comment by sharwell** [comment]
We regularly alter error messages and description pages for clarification in response to user feedback. Is there a specific situation where both:

1. There is evidence of a problematic/unclear error message and
2. There is a proposed alteration which, in the hands of users who experienced (1), is likely to lead to a proper resolution in less time?

Keep in mind that any significant alteration to an error message makes it _much_ more difficult for users to search online for solutions to that error (since existing help pages would all be written against the outdated message).
